datajoint/declare.py

```diff
@@ -414,6 +414,9 @@ def compile_attribute(line, in_key, foreign_key_sql):
     match['comment'] = match['comment'].replace('"', '\\"')   # escape double quotes in comment
     category, type_match = match_type(match['type'])
 
+    if match['comment'].startswith(':'):
+        raise DataJointError('An attribute comment must not start with a colon in comment "{comment}"'.format(**match))
```

Comment:
why is it that a comment cannot start with `:`? Does this mean something like
```
id:  int    #:my number
```
is not valid?

Reply:
That's right. We reserve comments that start with `:` to encode data types. We encode custom datatypes in the comment as `:custom-type: user comment`.

Reply:
OK got it and I recall agreeing to this strategy.